model/publishing/delivery/tasks/RemoteTaskStatusSynchroniser.php

```diff
@@ -33,21 +36,41 @@ public function getRemoteStatus()
      */
     public function __invoke($params)
     {
-        if (count($params) != 2) {
+        if (count($params) != 4) {
             throw new \common_exception_MissingParameter();
         }
         $taskId = array_shift($params);
         $envId = array_shift($params);
+        $testUri = array_shift($params);
+        $deliveryUri = array_shift($params);
 
-        $url = '/tao/TaskQueue/getStatus';
+        $url = '/tao/TaskQueue/get';
         $request = new Request('GET', trim($url, '/').'?'.http_build_query(['id' => $taskId]));
         $response = PlatformService::singleton()->callApi($envId, $request);
 
         if ($response->getStatusCode() == 200) {
             $body = json_decode($response->getBody()->getContents(), true);
             if ($body['success'] == true && isset($body['data'])) {
-                $this->status = $body['data'];
-                $report = new \common_report_Report(\common_report_Report::TYPE_SUCCESS, __('Status from remote successfully received.'), $this->status);
+                $this->status = $body['data']['status'];
+                $importAndCompileTaskReport = \common_report_Report::jsonUnserialize($body['data']['report']);
+                $remoteDeliveryId = 'Not found';
+
+                /** @var \common_report_Report $successReport */
+                foreach ($importAndCompileTaskReport->getSuccesses(true) as $successReport)
+                {
+                    \common_Logger::d($successReport->getMessage());
+                    if (
+                        strpos($successReport->getMessage(), 'QTI Test') !== false &&
```

Comment:
This is very ungly, unfortunately didn't find a better way filtering the deliveryId from the report.